app/cdap/services/helpers.js

```diff
@@ -906,6 +906,155 @@ const commaSeparatedNumber = (num) => {
+    'encryptor': 'icon-Encryptor',
+    'decryptor': 'icon-Decryptor',
+    'hdfsfilemoveaction': 'icon-filemoveaction',
+    'hdfsfilecopyaction': 'icon-filecopyaction',
+    'sqlaction': 'icon-SQLaction',
+    'impalahiveaction': 'icon-impalahiveaction',
+    'email': 'icon-emailaction',
+    'kinesissink': 'icon-Amazon-Kinesis',
+    'bigquerysource': 'icon-Big-Query',
+    'tpfsorc': 'icon-ORC',
+    'groupby': 'icon-groupby',
+    'sparkmachinelearning': 'icon-sparkmachinelearning',
+    'solrsearch': 'icon-solr',
+    'sparkstreaming': 'icon-sparkstreaming',
+    'rename': 'icon-rename',
+    'archive': 'icon-archive',
+    'wrangler': 'icon-DataPreparation',
+    'normalize': 'icon-normalize',
+    'xmlmultiparser': 'icon-XMLmultiparser',
+    'xmltojson': 'icon-XMLtoJSON',
+    'decisiontreepredictor': 'icon-decisiontreeanalytics',
+    'decisiontreetrainer': 'icon-DesicionTree',
+    'hashingtffeaturegenerator': 'icon-HashingTF',
+    'ngramtransform': 'icon-NGram',
+    'tokenizer': 'icon-tokenizeranalytics',
+    'skipgramfeaturegenerator': 'icon-skipgram',
+    'skipgramtrainer': 'icon-skipgramtrainer',
+    'logisticregressionclassifier': 'icon-logisticregressionanalytics',
+    'logisticregressiontrainer': 'icon-LogisticRegressionclassifier',
+    'hdfsdelete': 'icon-hdfsdelete',
+    'hdfsmove': 'icon-hdfsmove',
+    'windowssharecopy': 'icon-windowssharecopy',
+    'httppoller': 'icon-httppoller',
+    'window': 'icon-window',
+    'run': 'icon-Run',
+    'oracleexport': 'icon-OracleDump',
+    'snapshottext': 'icon-SnapshotTextSink',
+    'errorcollector': 'fa-exclamation-triangle',
+    'mainframereader': 'icon-MainframeReader',
+    'fastfilter': 'icon-fastfilter',
+    'trash': 'icon-TrashSink',
+    'staterestore': 'icon-Staterestore',
+    'topn': 'icon-TopN',
+    'wordcount': 'icon-WordCount',
+    'datetransform': 'icon-DateTransform',
+    'sftpcopy': 'icon-FTPcopy',
+    'sftpdelete': 'icon-FTPdelete',
+    'validatingxmlconverter': 'icon-XMLvalidator',
+    'wholefilereader': 'icon-Filereader',
+    'xmlschemaaction': 'icon-XMLschemagenerator',
+    's3toredshift': 'icon-S3toredshift',
+    'redshifttos3': 'icon-redshifttoS3',
+    'verticabulkexportaction': 'icon-Verticabulkexport',
+    'verticabulkimportaction': 'icon-Verticabulkload',
+    'loadtosnowflake': 'icon-snowflake',
+    'kudu': 'icon-apachekudu',
+    'orientdb': 'icon-OrientDB',
+    'recordsplitter': 'icon-recordsplitter',
+    'scalasparkprogram': 'icon-spark',
+    'scalasparkcompute': 'icon-spark',
+    'cdcdatabase': 'icon-database',
+    'cdchbase': 'icon-hbase',
+    'cdckudu': 'icon-apachekudu',
+    'changetrackingsqlserver': 'icon-database',
+    'conditional': 'fa-question-circle-o'
+  };
+
+  const pluginName = plugin ? plugin.toLowerCase() : '';
+  const icon = iconMap[pluginName] ? iconMap[pluginName]: 'fa-plug';
+  return icon;
+}
+
+const getNodesFromStages = (stages) => {
+  return stages.map(stage => {
+    let nodeInfo = {
```

Comment:
return {... };

instead of let nodeInfo = {... }; return nodeInfo;

Reply:
please fix

Reply:
Done